### DISCUSSION TITLE
How to create TAPP files

### DISCUSSION THREAD

**Discussion by Bournetoride** [arendst/Tasmota#17630] [Q&A] [answered]
Just starting on Berry with Beormund's Heating controller, trying to change .be files inside the TAPP is frustrating. 7zip will not save after I edit .be in Patformio. I can create a 7z archive from the required files ( with no compression using STORE option) but not able to change that to TAPP.  Should I address this to the 7zip team to add .TAPPas an accepted file type? Is there a better (windows) program to do this operation

**Comment by s-hadinger** [comment, marked as answer]
Did you try changing the extension from.tapp to.zip?

**Reply by sfromis** [reply]
IIRC, you should make sure that when updating the zip (before changing name back to tapp), it still becomes an **un**compressed zip.

**Reply by s-hadinger** [reply]
Uncompressed and without directories (flat files only)

**Reply by Bournetoride** [reply]
> Did you try changing the extension from .tapp to.zip?

yes but that simply created hc-host.zip.tapp  same when trying to rename new zip to .tapp
problems seems to be that .tapp is not a recognised extension even though 7zip will open them, it still cannot save as .tapp
I'm guessing that this was developed by Linux users

**Reply by Jason2866** [reply]
Rename the file from tapp to zip. Do your changes with 7zip. Rename the file from zip to tapp. Done.
Remove the unneeded extension. There has to be only one extension.

**Reply by Bournetoride** [reply]
Apologies, I didnt have the "show extensions" box ticked, its not easily visible in Win7
